torch/nn/modules/instancenorm.py

```diff
@@ -68,6 +70,16 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
     def forward(self, input: Tensor) -> Tensor:
         self._check_input_dim(input)
 
+        feature_dim = input.dim() - self._get_no_batch_dim()
+        if input.size(feature_dim) != self.num_features:
+            if self.affine:
+                raise ValueError(
+                    f"expected input's size at dim={feature_dim} to match num_features"
+                    f" ({self.num_features}), but got: {input.size(feature_dim)}.")
+            else:
+                warnings.warn(f"input's size at dim={feature_dim} does not match num_features "
+                              "(unused, since affine=False).")
```

Comment:
nit: maybe mention that the warning can be silenced by not passing `num_features` since it's unused anyway